CONTRIBUTING.md

```diff
@@ -1,13 +1,188 @@
+
+
+#### Project Management
+
+The cocotb project is [fairly popular](https://larsasplund.github.io/github-facts/verification-practices.html#frameworks) and the
+[gitter](https://gitter.im/cocotb/Lobby) and
+[Github issues](https://github.com/cocotb/cocotb) page receive a fair amount of traffic;
+which is only expected to increase.
+People are needed to categorize issues and pull requests, and respond to questions.
+Working this task is the quickest way to learn how cocotb works.
+Tending to this task requires the following:
+
+* people skills
+* an understanding of the scope of cocotb
+* general knowledge about how cocotb works
+
+To be able to add labels and close issues and PRs you will need special permissions.
+Contact a [maintainer](#maintainer) if you are interested in receiving these permissions.
+They will be granted according to need and the requestor's familiarity with cocotb.
+Please see the guidelines on [managing issues and pull requests](#Managing-of-Issues-and-Pull-Requests).
+
+Someone obliging this task should set notifications on the gitter channel to be notified of any new comments.
+They should also "watch" the Github repo by selecting the "Watching" notification level button in the upper right corner of the main Github page.
+Finally, they should notifying the maintainers they are able and willing to accept questions.
+
+This task can still be done without labeling and closing permissions.
+The best way to accomplish this is to comment on the issue or PR.
+This is especially helpful for Github issues about bugs.
+If you can duplicate the bug or confirm the bug report is invalid, that help maintainers *a lot*.
+
+
+#### Tests and Coverage
+
+Cocotb has a suite of unit tests (located in the `tests` directory) and examples (located in the `examples` directory) which are functional acceptance tests for Github pull requests.
+If pull requests cannot pass *all* of these tests, it  will likely be rejected.
+To ensure cocotb only includes the highest quality code, these test should be *very strenuous*.
+We use code coverage as a quantifiable metric of the "exhaustiveness" of these tests and want this metric to increase.
+
+Working this task requires a familiarity with
+
+* Cocotb's core functionality
+* How to write Verilog and VHDL
+* How to write cocotb tests in Python
+* (Optionally) Codecov; coverage agregator and Github bot
+* (Optionally) the [coverage]() module, for Python code coverage
+* (Optionally) [gcov](), for C++ code coverage
+* (Optionally) Travis CI and Github Actions, for automatic acceptance testing
+
+Cocotb's regression tests can be improved by
+
+* Testing more of cocotb's core functionality
+* Testing corner cases left out of the current set of tests (identifying them by looking at code coverage)
+* Increasing the matrix of simulators, operating system, and Python installations tested in CI
+
+Testing improvements don't require maintainer buy-in, but require a pull request.
```

Comment:
I mention "buy-in" a lot. Is there a better word? Should I add another section describing what that is, or maybe add it to the "How to Get Changes Merged" section and link it every time I mention it?

Reply:
Renamed to "pre-approval".